### PR TITLE
WINDUP-617: Create Java Classloader Report

### PR DESCRIPTION
**Pull request by jsight**
This is a very early snapshot of WIP on this:

https://issues.jboss.org/browse/WINDUP-617

There is no actual report yet. :) What is here is some basic support for specifying additional jar files as part of a technology (for example, specifying API jars from EAP) as well as support for detecting whether JDT was able to resolve a particular binding in the classpath.

Much more to come...

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/236/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/868/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/870/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/870/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/870/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutClassificationTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutClassificationTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/870/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/870/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/239/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/875/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>5</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutClassificationTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutClassificationTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/875/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/244/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/244/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/244/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/245/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/876/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/876/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/876/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li></ul>
Test FAILed.

**Comment by mbriskar** [comment]
I don't understand that classpath binding and classloader report. What is it's purpose?

**Comment by jsight** [comment]
retest

**Comment by jsight** [comment]
The most common initial problems when porting are missing classes due to changes in the environment. The purpose of this is to detect classes that were not resolvable in the input application as those will need to be addressed as part of the migration.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/249/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/880/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/250/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/881/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>6</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rule.groovy/GroovyExtensionJavaRulesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rule.groovy.GroovyExtensionJavaRulesTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/881/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/251/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/882/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>6</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rule.groovy/GroovyExtensionJavaRulesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rule.groovy.GroovyExtensionJavaRulesTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/882/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/253/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/253/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/253/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/884/<h2>Failed Tests: <span class='status-failure'>6</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>6</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rule.groovy/GroovyExtensionJavaRulesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rule.groovy.GroovyExtensionJavaRulesTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/884/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/256/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/887/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='org.jboss.windup.rules.apps:windup-rules-java-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/887/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport'>org.jboss.windup.rules.apps:windup-rules-java-tests</a>: <span class='status-failure'>2</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/887/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassSourceMatchTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassSourceMatchTest.testJavaClassCondition</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/887/org.jboss.windup.rules.apps$windup-rules-java-tests/testReport/org.jboss.windup.rules.java/JavaClassWithoutHintTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassWithoutHintTest.testJavaClassCondition</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/889/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/258/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/258/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/258/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/890/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/259/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/891/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/260/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/892/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/261/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/261/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/261/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/903/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/272/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/272/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/272/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

File: config-xml/addon/src/main/java/org/jboss/windup/config/parser/xml/PhaseHandler.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
  * 
  * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
  */
-@NamespaceElementHandler(elementName = "phase", namespace = "http://windup.jboss.org/schema/jboss-ruleset")
+@NamespaceElementHandler(elementName = "phase", namespace = RuleProviderHandler.WINDUP_RULE_NAMESPACE)
```
Comment (lincolnthree) [inline config-xml/addon/src/main/java/org/jboss/windup/config/parser/xml/PhaseHandler.java]: Excellent

**Pull request merged**